saleor/graphql/checkout/mutations/checkout_delivery_method_update.py

```diff
@@ -266,9 +267,24 @@ def _update_delivery_method(
         invalidate_prices_updated_fields = invalidate_checkout(
             checkout_info, lines, manager, save=False
         )
-        checkout.save(
-            update_fields=checkout_fields_to_update + invalidate_prices_updated_fields
-        )
+        try:
+            checkout.save(
```

Comment:
It depends on what we want to cover. If we want to cover a case when the object is deleted because of calling 2 mutations for the same thread (where one is checkoutComplete). Then we should have generic utils, which will make a lock before calling save on the object. If the object doesn't exist, then skip the actions.  This would require changes in the all places where we want to call a save on checkout with flow to handle a case where checkout doesn't exist anymore.
The second option that we could discuss is to clarify if calling multiple mutations for a single checkout is a proper way of handling it. Here we have a problem with `.save()` + `checkoutComplete`, but we could also have a situation where two different `.save()` + `.save()` are called at a similar time and produce strange results. For me, it sounds like a single checkout could be single-threaded

Reply:
I have adjusted the code to cover the first part of our needs. The second one will get a ticket and should be addressed in the future.